DrawerViewExample/DrawerViewExample/DrawerView.swift

```diff
@@ -0,0 +1,285 @@
+    self.peekView = peekView
+    self.contentView = contentView
+    super.init(frame: .zero)
+
+    addSubview(peekView)
+    setupPeekView()
+
+    addSubview(scrollingContentView)
+
+    scrollingContentView.backgroundColor = .white
+    scrollingContentView.topToBottom(of: peekView)
+    scrollingContentView.leftToSuperview()
+    scrollingContentView.rightToSuperview()
+    scrollingContentView.bottomToSuperview()
+
+    scrollingContentView.addSubview(contentView)
+    contentView.edgesToSuperview()
+    contentView.width(to: self)
+  }
+
+  private func setupPeekView() {
+    peekView.edgesToSuperview(excluding: .bottom)
+
+    let tapGestureRecogniser = UITapGestureRecognizer(target: self, action: #selector(filterDidTap))
+    peekView.addGestureRecognizer(tapGestureRecogniser)
+    peekView.addGestureRecognizer(panToExpandGestureRecogniser)
+    peekView.addGestureRecognizer(panToCollapseGestureRecogniser)
+    panToCollapseGestureRecogniser.isEnabled = false
+    peekView.isUserInteractionEnabled = true
+  }
+
+  public required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  @objc
+  func filterDidTap(_ gestureRecogniser: UITapGestureRecognizer) {
+    if isExpanded {
+      collapse()
+    } else {
+      expand()
+    }
+  }
+
+  public func expand() {
+    delegate?.filterViewWillExpand(self)
+    animatorForExpandGesture().startAnimation()
+  }
+
+  public func collapse() {
+    // this prevents us crashing under a circumstance which I can't remember
+    if animatingConstraint != nil {
+      animatorForCollapseGesture().startAnimation()
+    }
+  }
+
+  var upperAnimationLimit: CGFloat {
+    // This is a bit of a guess, could do with re-working. This prevents the filter going behind
+    // the navigation bar, which means you can never close the filters.
+    return -1 * min(scrollingContentView.contentSize.height, UIScreen.main.bounds.height - 300)
+  }
+
+  private var isExpanded: Bool {
+    return animatingConstraint.constant == upperAnimationLimit
+  }
+
+  @objc
+  func collapsedFilterDidPan(_ gestureRecogniser: UIPanGestureRecognizer) {
+    switch gestureRecogniser.state {
+    case .began:
+      startExpandingAnimation(fromGesture: gestureRecogniser)
+    case .changed:
+      animateExpanding(fromGesture: gestureRecogniser)
+    case .ended:
+      finishExpandingAnimation(fromGesture: gestureRecogniser)
+    default:
+      break
+    }
+  }
+
+  private func startExpandingAnimation(fromGesture gestureRecogniser: UIPanGestureRecognizer) {
+    guard let superView = superview else {
+      return
+    }
+    superView.setNeedsLayout()
+    superView.layoutIfNeeded()
+    initialTouchPosition = gestureRecogniser.translation(in: superView).y
+
+    delegate?.filterViewWillExpand(self)
+
+    propertyAnimator = animatorForExpandGesture()
+    propertyAnimator.startAnimation()
+    propertyAnimator.pauseAnimation()
```

Comment:
useful to know - immediately pausing a `UIViewPropertAnimator` puts it into a mode which you can then change it manually via the `fractionComplete` property - which is done in `animateExpanding(fromGesture:)`

Reply:
this then allows you to have an animation which can be performed as a standard animator, or one that the user can interact with!